src/marqo/s2_inference/s2_inference.py

```diff
@@ -27,19 +27,33 @@ def vectorise(model_name: str, content: Union[str, List[str]], device: str = get
     Raises:
         VectoriseError: if the content can't be vectorised, for some reason.
     """
+
+    model_name = _get_model_name(model)
+
     model_cache_key = _create_model_cache_key(model_name, device)
 
-    if model_cache_key not in available_models:
-        available_models[model_cache_key] = _load_model(model_name, device=device)
-        logger.info(f'loaded {model_name} on device {device} with normalization={normalize_embeddings}')
+    _update_model_properties(model, model_name)
+
+    _update_available_models(model_cache_key, model_name, device, normalize_embeddings)
+
     try:
         vectorised = available_models[model_cache_key].encode(content, normalize=normalize_embeddings, **kwargs)
     except UnidentifiedImageError as e:
         raise VectoriseError from e
 
     return _convert_vectorized_output(vectorised)
 
-def _create_model_cache_key(model_name: str, device: str) -> Tuple:
+
+def _get_model_name(model: Union[str, dict]) -> str:
+    """returns the name of the model
+    """
+    if isinstance(model, str):
+        return model
+    elif isinstance(model, dict):
+        return model['name']
```

Comment:
i think we want the check for the valid dict to happen before this in case the "name" field is not present